### PR TITLE
Turn off circleci linter

### PR DESCRIPTION
**Pull request by howardjohn**
We have moved this over to prow.

Note: Do not cherry pick this to release branches without also enabling the prow one on the release branch

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -372,15 +372,15 @@ jobs:
...
-  lint:
-    <<: *defaults
-    environment:
-      KUBECONFIG: /go/src/istio.io/istio/.circleci/config
-    resource_class: xlarge
-    steps:
-      - checkout
-      - run: make sync
-      - run: make lint
+#  lint:
```
Comment (kyessenov) [inline .circleci/config.yml]: I think we generally remove code rather than commenting it out. We have the git history anyways.

Comment (howardjohn) [reply .circleci/config.yml]: Agree, I was following the approach taken by @rshriram when he disabled the other tests. I can delete all the commented out though if you prefer

Comment (sdake) [reply .circleci/config.yml]: agreed please remove - we can revert if needed.

Comment (howardjohn) [reply .circleci/config.yml]: sgtm, removed them all

**Comment by icygalz** [comment]
/lgtm

thanks! :)

**Comment by rshriram** [comment]
Merging. I see no reason why prow should fail for circleci config changes..

**Pull request merged**